### PR TITLE
Column streaming for differential assembly

### PR DESCRIPTION
**Pull request by hmusta**

### CODE REVIEW

**Comment by karasikov** [comment]
Is this ready for review?

**Comment by hmusta** [comment]
> Is this ready for review?

Yes

File: metagraph/src/graph/annotated_dbg.hpp [hunk 1]
```diff
@@ -61,7 +71,8 @@ class AnnotatedSequenceGraph {
 
   protected:
     std::shared_ptr<SequenceGraph> graph_;
-    std::unique_ptr<Annotator> annotator_;
+    std::unique_ptr<Annotation> annotation_;
+    const std::shared_ptr<Annotator> annotator_;
```
Comment (karasikov) [inline metagraph/src/graph/annotated_dbg.hpp:75]: Will it work if you pass lazy annotator to AnnotatedDBG? This points to a design issue.
Maybe it's better to do without AnnotatedDBG in differential assembly? Better pass graph and annotation separately. Then you can pass lazy annotation in any type you want.

**Review by karasikov — CHANGES_REQUESTED** [review]

**Comment by hmusta** [comment]
Superseded by #487

**Pull request closed without merging**